packages/react/src/analytics/context.ts

```diff
@@ -12,26 +13,46 @@ export type WebContextType = {
       title: string;
       referrer: string;
     };
+    pageLocationReferrer: string | undefined;
   };
 };
 
+let lastLocation =
+  typeof document !== 'undefined' ? document.referrer : undefined;
+
 /**
- * Adds web environment information to the context.
+ * Returns the web context for the analytics package.
+ *
+ * @param trackEventType - The type of the event being tracked.
  *
  * @returns Context object for web applications.
  */
-export const context = (): WebContextType => {
+export const context = (
+  trackEventType?: ExtendedTrackTypes,
+): WebContextType => {
+  const pageLocationReferrer = lastLocation;
+  const locationHref = window.location.href;
+  const isPageEvent = trackEventType === trackTypes.PAGE;
+
+  if (isPageEvent && lastLocation !== locationHref) {
```

Comment:
Is there a case where the location changes but the event type is not page? What I mean is, why do we need to include the `isPageEvent` for this calculation? Wouldn't this be enough?

```suggestion
  if (lastLocation !== locationHref) {
```

Reply:
We must ensure this variable is only updated when a page view occurs, otherwise we would end up with page action events being sent with the incorrect location referrer, as this variable would be updated to the current page URL instead of holding the previous one until a new page view gets triggered.

Reply:
I understand the problem but I think that this should not be a concern for our context / package to have. If there is an event that is tracked in a new page before the page event is tracked, it would mean the context would register the `pageLocationReferrer` with an incorrect url. I think the context should not concern with that and it should be the responsibility of the application to make sure that a page event is triggered before an event is triggered when a new page is rendered. What do you think?

Reply:
Updated: Removed dependency of eventType from context.